### ISSUE TITLE
Converting stable diffusion 2.0's ckpt file to diffusers' format

### ISSUE THREAD

**Issue by larme** [huggingface/diffusers#4000]
### Describe the bug

Cannot convert stable diffusion 2.0's ckpt file into diffusers format using `convert_original_stable_diffusion_to_diffusers.py`

### Reproduction

Assuming you have used the stable diffusion 2.0 pipeline before, you can just run the following commands to reproduce the error:

```
python convert_original_stable_diffusion_to_diffusers.py \
       --checkpoint_path ~/.cache/huggingface/hub/models--stabilityai--stable-diffusion-2/snapshots/1e128c8891e52218b74cde8f26dbfc701cb99d79/768-v-ema.ckpt \
       --original_config_file ~/codes/stablediffusion/configs/stable-diffusion/v2-inference.yaml \
       --dump_path ~/tmp/sd2 \
       --prediction_type v_prediction \
       --image_size 768
```

where `~/codes/stablediffusion` is a copy of https://github.com/stability-AI/stablediffusion/

### Logs

```shell
the entire log is here:

https://gist.github.com/larme/f348b35780da3d2f763d66abc2251433
```

### System Info

- `diffusers` version: 0.18.0.dev0
- Platform: Linux-6.1.0-9-amd64-x86_64-with-glibc2.35
- Python version: 3.10.6
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.30.2
- Accelerate version: not installed
- xFormers version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@patrickvonplaten @say

**Comment by cmdr2** [comment]
Ran into this bug as well. Seems to be coming from this line (which changes the CLIP text_model used) - https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py#L888

**Comment by cmdr2** [comment]
I can reproduce this as well, seems to be from this change: https://github.com/huggingface/diffusers/commit/bc9a8cef6f258aafcd43ef64ac161218a7eae43a#diff-0613bbbcc958a37615e7638e09307201784c90c2661611f745fe4f67f68f51efL863

**Comment by patrickvonplaten** [comment]
Thanks for the issue, looking into it!

**Comment by patrickvonplaten** [comment]
Hey @cmdr2,

Very sorry about this, you were 100% correct. I've corrected everything and just tested in with #4041. 

Please let me know if it isn't fixed.

**Comment by cmdr2** [comment]
Thanks @patrickvonplaten for the quick fix!

Tested that it works well for me now.

No problem, and sorry I didn't feel confident enough about my changes to propose a patch.
